src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpMetadataContext.cs

```diff
@@ -9,27 +9,38 @@ namespace Microsoft.CodeAnalysis.CSharp.ExpressionEvaluator
     internal struct CSharpMetadataContext
     {
         internal readonly ImmutableArray<MetadataBlock> MetadataBlocks;
+        internal readonly ImmutableArray<Alias> Aliases;
         internal readonly CSharpCompilation Compilation;
         internal readonly EvaluationContext EvaluationContext;
 
-        internal CSharpMetadataContext(ImmutableArray<MetadataBlock> metadataBlocks, CSharpCompilation compilation)
+        internal CSharpMetadataContext(
+            ImmutableArray<MetadataBlock> metadataBlocks, 
+            ImmutableArray<Alias> aliases,
```

Comment:
I understood the ```EvaluationContext``` was intended to encapsulate all the state we received from the debugger.  If that's not the case, we could conceivably recomputed the aliases on every API call and pass them in to each individual ```EvaluationContext``` method.

Alternatively, if the caching proves to be problematic, we could partially reuse the ```EvaluationContext``` when the aliases don't match (e.g. the way we currently reuse only the ```Compilation``` in some cases).